support/util/pausereconcile_test.go

```diff
@@ -0,0 +1,52 @@
+package util
+
+import (
+	"testing"
+	"time"
+
+	. "github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+)
+
+func TestIsReconciliationPaused(t *testing.T) {
+	testsCases := []struct {
+		name             string
+		inputAnnotations map[string]string
+		expectedResult   bool
+	}{
+		{
+			name:             "if pausedReconciliation annotation does not exist then reconciliation is not paused",
+			inputAnnotations: map[string]string{},
+			expectedResult:   false,
+		},
+		{
+			name: "if pausedReconciliation annotation is later than time.Now then reconciliation is paused",
+			inputAnnotations: map[string]string{
+				hyperv1.PauseReconciliationAnnotation: time.Now().Add(4 * time.Hour).Format(time.RFC3339),
+			},
+			expectedResult: true,
+		},
+		{
+			name: "if pausedReconciliation annotation is before time.Now then reconciliation is not paused",
+			inputAnnotations: map[string]string{
+				hyperv1.PauseReconciliationAnnotation: time.Now().Add(-4 * time.Hour).Format(time.RFC3339),
+			},
+			expectedResult: false,
+		},
+		{
+			name: "if pausedReconciliation annotation is true then reconciliation is paused",
+			inputAnnotations: map[string]string{
+				hyperv1.PauseReconciliationAnnotation: "true",
+			},
```

Comment:
should have a case for wrong annotation value, what happens then?

Reply:
Wrong annotation value just isn't accepted and it's treated as if the label was never specified. We can log that though

Reply:
>Wrong annotation value just isn't accepted and it's treated as if the label was never specified. We can log that though

Ok make sense, but so a wrong value is a possible input so I won't believe you unless there's test case that proves IsReconciliationPaused=false for that input :)

Reply:
adding test case

Reply:
added